### ISSUE TITLE
Swagger2 spec validator

### ISSUE THREAD

**Issue by ikitommi** [metosin/spec-swagger#7]
Should be available for Clojure in a separate ns just like it's in ring-swagger.

```clj
(require '[spec-swagger.swagger2.validator :as v])

(v/validate {:swagger "2.0"})
; ({:level "error",
;   :schema {:loadingURI "#", :pointer ""},
;   :instance {:pointer ""},
;   :domain "validation",
;   :keyword "required",
;   :message "object has missing required properties ([\"info\",\"paths\"])",
;   :required ["info" "paths" "swagger"],
;   :missing ["info" "paths"]})
```

**Comment by miikka** [comment]
Why not just use the one in ring-swagger?

**Comment by ikitommi** [comment]
yes. As long as these are separate libs, would be cool not to need to
depend on ring-swagger from spec-side. Lots of deps. ring-swagger 1.0.0
could be a modularized version supporting both schema & spec, as modules,
some how... is this sound?

**Comment by miikka** [comment]
Well, it should be easy enough to split out the validator from ring-swagger. It adds a bit of maintenance burden, but on the other hand, I think the validator has been quite stable compared to the other parts of ring-swagger.

**Comment by ikitommi** [comment]
ok, so this could be the plan:
1) copy-paste the validator here to make spec-swagger independent for now.
2) modularize/split ring-swagger into parts when spec-swagger is "done". Parts could be something like `core`, `routes`, `validator`, `schema`, `spec` and a `all`. Like Ring.
